File: 3869-smallest-index-with-digit-sum-equal-to-index/smallest-index-with-digit-sum-equal-to-index.java
class Solution {
    public int smallestIndex(int[] nums) {
        int sum=0;
        for(int i=0;i<nums.length;i++){
            int num=nums[i];
            while(num>0){
                sum+=num%10;
                num/=10;
            }
            if(sum==i){
                return i;
            }
            sum=0;
        }
        return -1;
    }
}

File: 3869-smallest-index-with-digit-sum-equal-to-index/smallest-index-with-digit-sum-equal-to-index.c
int smallestIndex(int* nums, int numsSize) {
    for(int i=0;i<numsSize;i++){
        int sum=0;
        int num=*nums;
        while(num>0){
            int digit=num%10;
            sum+= digit;
            num/=10;   
        }
        if(i==sum){
            return i;
        }
        nums++;
    }
    return -1;
}